contracts/staking/test/unit_tests/lib_proxy.ts

```diff
@@ -0,0 +1,324 @@
+import { blockchainTests, constants, expect, hexRandom } from '@0x/contracts-test-utils';
+import { StakingRevertErrors } from '@0x/order-utils';
+
+import { artifacts, TestLibProxyContract, TestLibProxyReceiverContract } from '../../src';
+
+blockchainTests.resets('LibProxy', env => {
+    let proxy: TestLibProxyContract;
+    let receiver: TestLibProxyReceiverContract;
+
+    // Generates a random bytes4 value.
+    function randomBytes4(): string {
+        return hexRandom(4);
+    }
+
+    before(async () => {
+        proxy = await TestLibProxyContract.deployFrom0xArtifactAsync(
+            artifacts.TestLibProxy,
+            env.provider,
+            env.txDefaults,
+            artifacts,
+        );
+
+        receiver = await TestLibProxyReceiverContract.deployFrom0xArtifactAsync(
+            artifacts.TestLibProxyReceiver,
+            env.provider,
+            env.txDefaults,
+            artifacts,
+        );
+    });
+
+    enum RevertRule {
+        RevertOnError,
+        AlwaysRevert,
+        NeverRevert,
+    }
+
+    // Choose a random 4 byte string of calldata to send and prepend with `0x00` to ensure
+    // that it does not call `externalProxyCall` by accident. This calldata will make the fallback
+    // in `TestLibProxyReceiver` fail because it is 4 bytes long.
+    function constructRandomFailureCalldata(): string {
+        return '0x00'.concat(randomBytes4().slice(4, 10));
```

Comment:
Seems clearer to just do:
```suggestion
        return hexConcat('0x00', hexRandom(3));
```

Reply:
Or maybe even:
```suggestion
        return hexLeftPad(hexRandom(3), 4);
```